openssl/src/base64.rs

```diff
@@ -0,0 +1,170 @@
+use std::ptr;
+use std::slice;
+
+use ffi;
+use error::ErrorStack;
+use {cvt_p, cvt};
+
+struct Bio(*mut ffi::BIO);
+
+impl Bio {
+    pub fn as_ptr(&self) -> *mut ffi::BIO {
+        self.0
+    }
+}
+
+impl Drop for Bio {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::BIO_free_all(self.0);
+        }
+    }
+}
+
+/// Encode `data` in base64. If `newlines` is true then the data is
+/// encoded with line breaks.
+pub fn encode(data: &[u8], newlines: bool) -> Result<Vec<u8>, ErrorStack> {
```

Comment:
I'd make an `enum Newline { Yes, No }` rather than passing in a boolean here - it's not super clear what `base64::encode(data, true)` means.